src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -108,11 +108,13 @@ public ClientCertificateOption ClientCertificateOptions
                     {
                         case ClientCertificateOption.Manual:
                             ThrowForModifiedManagedSslOptionsIfStarted();
-                            _socketsHttpHandler.SslOptions.LocalCertificateSelectionCallback = null;
+                            _clientCertificateOptions = value;
+                            _socketsHttpHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate(ClientCertificates);
```

Comment:
There is probably a better way to pass in the 'ClientCertificates' collection into this delegate.  This current call probably allocates extra for the closure.  Suggestions welcome.

Reply:
Unfortunately I think this is about as good as can be done, since there's no state object with the delegate that allows arbitrary state to be passed through.  It could be worse: the compiler could need to allocate both a closure and a delegate to the closure.  In this case, though, the only state being closed over is `this` in order to access `this.ClientCertificates`, so no closure is needed: the compiler will generate the lambda as an instance method, and then it'll allocate a delegate that points to that instance method.  So there will still be an allocation for the delegate, but at least not two.  Since this should generally only happen once per HttpClientHandler, I think it's ok.

Reply:
ok.  Thx for the detailed explanation.